src/components/connect.js

```diff
@@ -47,7 +37,6 @@ export default function connect(mapStateToProps, mapDispatchToProps, mergeProps,
 
   const finalMergeProps = mergeProps || defaultMergeProps
   const { pure = true, withRef = false } = options
-  const checkMergedEquals = pure && finalMergeProps !== defaultMergeProps
```

Comment:
@tgriesser can you take a look at that plz? I've been able to make the tests pass without using this variable. Is there a missing test-case?